Mathlib/NumberTheory/Cyclotomic/PrimitiveRoots.lean

```diff
@@ -432,7 +426,8 @@ theorem sub_one_norm_prime [hpri : Fact (p : ℕ).Prime] [hcyc : IsCyclotomicExt
 -- Porting note: writing `(2 : ℕ+)` was not needed (similarly everywhere).
 theorem pow_sub_one_norm_two {k : ℕ} (hζ : IsPrimitiveRoot ζ (2 ^ (k + 1)))
     [IsCyclotomicExtension {(2 : ℕ+) ^ (k + 1)} K L]
-    (hirr : Irreducible (cyclotomic (2 ^ (k + 1)) K)) : norm K (ζ ^ 2 ^ k - 1) = (-2) ^ 2 ^ k := by
+    (hirr : Irreducible (cyclotomic (2 ^ (k + 1)) K)) :
+    norm K (ζ ^ 2 ^ k - 1) = (-2 : K) ^ 2 ^ k := by
```

Comment:
I am not sure I understand the point, why `(-2 : K)` is better?

Reply:
The previous code parsed as `(((-2) ^ 2 ^ k : ℤ) : K)`, which didn't match mathlib3